### DISCUSSION TITLE
HDR10+ info not applying

### DISCUSSION THREAD

**Discussion by jriker1** [quietvoid/hdr10plus_tool#65] [Q&A]
I extracted from a 4K HDR10/HDR10+ video HDR10+ info with:

`ffmpeg -i input.mkv -map 0:v:0 -c copy -vbsf hevc_mp4toannexb -f hevc - | hdr10plus_tool extract -o metadata.json -`

The code I used ti put it back in without an additional step is:

`ffmpeg -i input.mkv -sn -an -f yuv4mpegpipe -strict -1 -pix_fmt yuv420p10le - | x265-10b - --input-depth 10 --output-depth 10 --y4m --preset slow --hdr10 --hdr10-opt --high-tier --repeat-headers --crf 17 --master-display "G(13250,34500)B(7500,3000)R(34000,16000)WP(15635,16450)L(40000000,50)" --max-cll "1000,401" --chromaloc 2 --no-sao --range limited --keyint 24 --colormatrix bt2020nc --colorprim bt2020 --transfer smpte2084 --dhdr10-info=metadata.json --vbv-bufsize 160000 --vbv-maxrate 160000 <output>.hevc`

When done there is no HDR10+ info and MediaInfo doesn't see it.  If I try injecting it with 

`hdr10plus_tool inject -i video.hevc -j metadata.json -o injected_output.hevc`

it says:

```
Parsing JSON file...
Processing input video for frame order info...

Warning: Input file already has HDR10+ SEIs, they will be replaced.
```

So hdr10plus_tool seems to "think" something is there but it's definitely not thru mediainfo or ffprobe.  Does show after injecting it afterwards separately.

Not sure if this is an ffmpeg issue or something you are aware of with dhdr10-info maybe needs to be adjusted?

**Comment by quietvoid** [comment]
If you can upload a small sample of your file then I can check if MediaInfo detects it on my side.

**Reply by jriker1** [reply]
It's a 2 hour blueray is that even possible and would the HDR10/HDR10+ be retained?  Was going to upload the JSON but isn't accepted in the forum.

Note when running ffmpeg only reference I noticed to the json file was this:

`x265 [info]: tools: deblock dhdr10-info`

**Reply by quietvoid** [reply]
There should be metadata for every frame so you should be able to use `mkvmerge` to trim at most 100 frames.
